src/middlewared/middlewared/plugins/service.py

```diff
@@ -807,14 +807,34 @@ def _initializing_smartd_pid(self):
             asyncio.ensure_future(self.restart('collectd'))
 
     async def _restart_ups(self, **kwargs):
+        await self.middleware.call('service._restart_ups_job')
+
+    @job(lock='restart_upsmon')
+    async def _restart_ups_job(self, job, **kwargs):
```

Comment:
service.restart is supposed to return a boolean whether the service started or not.

Now it wont be possible since we wont wait for it?

Reply:
Yes, i am afraid that is a drawback using this approach. We can't trust service.restart anymore for ups.restart. After discussing with @themylogin we decided job is the best way to accomplish this because of the varying time upsmon might take to fully stop. Please let me know if you think we should handle this in another way

Reply:
Hm, why? How about it kick it with a big hammer? Wait a few seconds, if it didnt stop, kill -9 everything.

Reply:
Perfect :D It raises a upssched event by spawning a new process for it, so i guess we can wait for 15 seconds and if still it is going on, kick it with a big hammer :P ?

Reply:
I will say 10, unless you can think anything important its doing that grants a clean shutdown of the process? I cant

Reply:
Well the only thing i have observed is the upssched event in which we also send a mail. So i think we can have it 10 for now, and if there are any issues later on, we can revisit this. Will have the changes up for your review soon. Thank you